### DISCUSSION TITLE
Create user-defined fields in the contacts & sync via Activesync

### DISCUSSION THREAD

**Discussion by gueni222** [tine20/tine20#7444] [Q&A]
Hello all,

I don't know if this is the right place.
I can save custom phone numbers in my Kontakze app.
These are not saved in the contacts or are deleted when I change and save something in the Tine contacts.
Is it possible to create user-defined fields in Tine and are these then synchronised via Activesync?

Thank you very much for any solutions or suggestions.
Many greetings
Günter

**Comment by lab-at-nohl** [comment]
I don't know what contacts app you use. Mine has also a lot of possible phone numbers, which I don't use. And I am not sure about ActiveSync but I have never seen custom fields. Sorry. Use description text instead. 

To find out, just fill in every field in Tine contact with a number and see what's synced and what's not. I would guess phone, fax and mobile...

**Comment by gueni222** [comment]
Hello,

the contact app is called True Phone and sends for example Custom Fields with a featured x via Activesync.
The only question is if the fields can be passed transparently on Tine.
So they would also be a backup for the particular device.

Example part of the contact as VCard

BEGIN:VCARD
VERSION:2.1
TEL;WORK;PREF:+4933204200
TEL;X-Sozialdienst:+493320421229
TEL;CELL:+4917610337446
TEL;X-Station-F5:+493320422255
NOTE:Example note
END:VCARD

Greetings
Günter 


![Lumii_20230512_231044593.jpg](https://github.com/tine20/tine20/assets/50445885/84a930d0-b6ba-4ea0-9051-548daf4e4856)

**Comment by lab-at-nohl** [comment]
Are you sure it is ActiveSync? Not cardDAV?

For cardDAV unknown clients are read-only. It is not to complicated to program a mapping for a new cardDAV client to make it syncing in both directions. However, I do not see a way to include custom types for phone numbers. Usually number types like home, business or mobile are fixed. 

I am not aware of any mapping to custom fields for any client. Although the idea is good, I do not see a broad use case - most clients do not offer custom types. Or am I wrong?

**Comment by gueni222** [comment]
Hello,

Well, in Google Account I can also set up custom fields.
The fields are also transferred from the other account and syncronised in the Google account. I don't know which protocol is used.
I once installed a SOGo groupware where the syncronisation via Activsync also worked with these transparent fields with the "X" in front of them.
I find it practical to have a company with the numbers for sales, logistics, support, etc. and not always new contacts for each contact person.
Or, as in my example, the hospital Beelitz Heilstättene.

Greetings
Günter

**Reply by lab-at-nohl** [reply]
You are right, I see. Other providers do offer custom fields for contacts. And custom dates. 

However, ActiveSync definitely uses fixed field names like `HomePhoneNumber` or `MobilePhoneNumber`. See [MS-ASCNTC](https://interoperability.blob.core.windows.net/files/MS-ASCNTC/%5bMS-ASCNTC%5d.pdf) for a overview of fields on page 4. Thus, I am pretty sure we are talking about DAV respectively cardDAV for contacts. 

You could add a feature request by opening a new issue. But I don't know how to generalize a use case that works for everybody. The program logic would be something like this: Allow cardDAV clients to sync against Tine's custom fields. But how to identify the specific field? If I get you right, you want to create the custom field on the phone and its name can change...

I would use several contacts with the same company as you already do. 
Maybe, someone has an idea?

Best regards
Johannes
